From f0e647e6a013175f8ba80057f6db22b016534a76 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:40 -0400
Subject: [powerpc] Use 32 bit loads and stores when operating on condition register values

Message-id: <1410545655-205645-172-git-send-email-dzickus@redhat.com>
Patchwork-id: 94274
O-Subject: [RHEL7 PATCH 171/626] powerpc: Use 32 bit loads and stores when operating on condition register values
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit bbe30b3b576b4ba96641a7c244071282618b2cbb
Author: Anton Blanchard <anton@samba.org>
Date:   Tue Oct 15 14:36:31 2013 +1100

    powerpc: Use 32 bit loads and stores when operating on condition register values

    The condition register (CR) is a 32 bit quantity so we should use
    32 bit loads and stores.

    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/tm.S b/arch/powerpc/kernel/tm.S
index c8c58b3..5752ace 100644
--- a/arch/powerpc/kernel/tm.S
+++ b/arch/powerpc/kernel/tm.S
@@ -106,7 +106,7 @@ DSCR_DEFAULT:
 _GLOBAL(tm_reclaim)
  mfcr r6
  mflr r0
- std r6, 8(r1)
+ stw r6, 8(r1)
  std r0, 16(r1)
  std r2, 40(r1)
  stdu r1, -TM_FRAME_SIZE(r1)
@@ -285,7 +285,7 @@ dont_backup_fp:
  REST_NVGPRS(r1)
 
  addi    r1, r1, TM_FRAME_SIZE
- ld r4, 8(r1)
+ lwz r4, 8(r1)
  ld r0, 16(r1)
  mtcr r4
  mtlr r0
@@ -310,7 +310,7 @@ dont_backup_fp:
 _GLOBAL(__tm_recheckpoint)
  mfcr r5
  mflr r0
- std r5, 8(r1)
+ stw r5, 8(r1)
  std r0, 16(r1)
  std r2, 40(r1)
  stdu r1, -TM_FRAME_SIZE(r1)
@@ -444,7 +444,7 @@ restore_gprs:
  REST_NVGPRS(r1)
 
  addi    r1, r1, TM_FRAME_SIZE
- ld r4, 8(r1)
+ lwz r4, 8(r1)
  ld r0, 16(r1)
  mtcr r4
  mtlr r0
diff --git a/arch/powerpc/platforms/powernv/opal-wrappers.S b/arch/powerpc/platforms/powernv/opal-wrappers.S
index 2a03e1e..403d058 100644
--- a/arch/powerpc/platforms/powernv/opal-wrappers.S
+++ b/arch/powerpc/platforms/powernv/opal-wrappers.S
@@ -24,7 +24,7 @@
  mflr r0;   \
  mfcr r12;   \
  std r0,16(r1);  \
- std r12,8(r1);  \
+ stw r12,8(r1);  \
  std r1,PACAR1(r13);  \
  li r0,0;   \
  mfmsr r12;   \
@@ -53,7 +53,7 @@ _STATIC(opal_return)
   */
  FIXUP_ENDIAN
  ld r2,PACATOC(r13);
- ld r4,8(r1);
+ lwz r4,8(r1);
  ld r5,16(r1);
  ld r6,PACASAVEDMSR(r13);
  mtspr SPRN_SRR0,r5;
-- 
1.7.1